go/stellar/stellarsvc/service.go

```diff
@@ -296,3 +310,44 @@ func (s *Server) FormatLocalCurrencyString(ctx context.Context, arg stellar1.For
 
 	return res, nil
 }
+
+// check that the display amount is within 1% of current exchange rates
+func (s *Server) checkDisplayAmount(ctx context.Context, arg stellar1.SendLocalArg) error {
+	if arg.DisplayAmount == "" {
+		return nil
+	}
+
+	exchangeRate, err := remote.ExchangeRate(ctx, s.G(), arg.DisplayCurrency)
+	if err != nil {
+		return err
+	}
+
+	xlmAmount, err := stellar.ConvertLocalToXLM(arg.DisplayAmount, exchangeRate)
+	if err != nil {
+		return err
+	}
+
+	currentAmt, err := amount.ParseInt64(xlmAmount)
+	if err != nil {
+		return err
+	}
+
+	argAmt, err := amount.ParseInt64(arg.Amount)
+	if err != nil {
+		return err
+	}
+
+	if percentageAmountChange(currentAmt, argAmt) > 1.0 {
+		return errors.New("current exchange rates have changed more than 1%")
+	}
+
+	return nil
+}
+
+func percentageAmountChange(a, b int64) float64 {
+	if a == 0 && b == 0 {
+		return 0.0
+	}
+	mid := 0.5 * float64(a+b)
+	return math.Abs(100.0 * float64(a-b) / mid)
```

Comment:
I'm on the lookout for float math on money. But don't have concrete rules. Is there a reason this float math is safe? exchange.go has some big rational stuff.